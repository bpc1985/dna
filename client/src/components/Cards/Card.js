import React from "react";
import "./Card.scss";

export default function Card({subscription, onDetail}) {
  const { _id, phone_number, address, dnaPackage } = subscription;
  return (
    <li className="subscriptionItem">
      <div>
        <h1>{dnaPackage.name}</h1>
        <h2>
          {dnaPackage.type === 'mobile' ? phone_number : address}
        </h2>
      </div>
      <div>
        <span>{dnaPackage.type}</span>
        <button className="btn" onClick={() => onDetail(_id)}>
          Details
        </button>
      </div>
    </li>
  );
}
