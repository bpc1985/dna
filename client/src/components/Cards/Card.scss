@import "stylesheets/index";

.subscriptionItem {
  margin: 1rem 0;
  padding: 1rem;
  border: 1px solid $pink;
  display: flex;
  justify-content: space-between;
  align-items: center;

  h1 {
    margin: 0;
    font-size: 1rem;
    color: $pink;
  }

  h2 {
    margin: 0;
    font-size: 1rem;
    color: $dark-gray;
  }

  p {
    margin: 0;
  }

  span {
    margin-right: 10px;
    font-size: 0.8rem;
    color: $dark-gray;
  }

  .btn {
    background: $pink;
    font: inherit;
    border: 1px solid $pink;
    border-radius: 3px;
    padding: 0.25rem 1rem;
    box-shadow: 1px 1px 5px rgba(0, 0, 0, 0.26);
    color: $white;
    cursor: pointer;

    &:hover, &:active {
      background: $purple;
      border-color: $purple;
    }
  }
}

