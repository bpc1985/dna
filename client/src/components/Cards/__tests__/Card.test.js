import { render, fireEvent } from "@testing-library/react";
import React from "react";
import Card from "../Card";

const subscription = {
  _id: "111",
  phone_number: "111111111",
  address: "Address 1",
  dnaPackage: {
    name: "Package 1",
    description: "Description 1",
    type: "mobile"
  }
};

describe("Card", () => {
  it("should render component", () => {
    const onDetail = jest.fn();
    const utils = render(<Card subscription={subscription} onDetail={onDetail} />);
    expect(utils.container).toMatchSnapshot();
    const button = utils.container.querySelector('button');
    fireEvent.click(button);
    expect(onDetail).toHaveBeenCalled();
  });
});
