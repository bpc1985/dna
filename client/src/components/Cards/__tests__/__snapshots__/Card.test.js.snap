// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render component 1`] = `
<div>
  <li
    class="subscriptionItem"
  >
    <div>
      <h1>
        Package 1
      </h1>
      <h2>
        111111111
      </h2>
    </div>
    <div>
      <span>
        mobile
      </span>
      <button
        class="btn"
      >
        Details
      </button>
    </div>
  </li>
</div>
`;
